src/app/games/rectanglz/tests/rectanglz.component.spec.ts

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable max-lines-per-function */
+import { fakeAsync } from '@angular/core/testing';
+import { ComponentTestUtils } from 'src/app/utils/tests/TestUtils.spec';
+import { RectanglzComponent } from '../rectanglz.component';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+import { RectanglzState } from '../RectanglzState';
+import { RectanglzConfig, RectanglzRules } from '../RectanglzRules';
+import { RectanglzMove } from '../RectanglzMove';
+import { Coord } from 'src/app/jscaip/Coord';
+import { PlayerOrNone } from 'src/app/jscaip/Player';
+import { RectanglzFailure } from '../RectanglzFailure';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+
+describe('RectanglzComponent', () => {
+
+    const _: PlayerOrNone = PlayerOrNone.NONE;
+    const O: PlayerOrNone = PlayerOrNone.ZERO;
+    const X: PlayerOrNone = PlayerOrNone.ONE;
+    const defaultConfig: MGPOptional<RectanglzConfig> = RectanglzRules.get().getDefaultRulesConfig();
+
+    let testUtils: ComponentTestUtils<RectanglzComponent>;
+
+    beforeEach(fakeAsync(async() => {
+        testUtils = await ComponentTestUtils.forGame<RectanglzComponent>('Rectanglz');
+    }));
+
+    it('should create', () => {
+        testUtils.expectToBeCreated();
+    });
+
+    describe('first click', () => {
+
+        it('should select piece when clicking on it', fakeAsync(async() => {
+            // Given any state
+            // When clicking on one of your piece
+            await testUtils.expectClickSuccess('#click_0_0');
+
+            // Then it should have selected the piece
+            testUtils.expectElementToHaveClass('#piece_0_0', 'selected-stroke');
+        }));
+
+        it('should toast when selecting opponent piece', fakeAsync( async() => {
+            // Given any state
+            // When clicking on an opponent piece
+            // Then the move should be illegal
```

Comment:
c'est quoi notre format ici, c'est pas "Then it should fail"'?

Reply:
Plus que jamais !